platform/sponge/src/main/java/me/lokka30/treasury/plugin/sponge/TreasurySponge.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/ArcanePlugins/Treasury>.
+ */
+
+package me.lokka30.treasury.plugin.sponge;
+
+import com.google.inject.Inject;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import me.lokka30.treasury.api.common.event.EventExecutorTrackerShutdown;
+import me.lokka30.treasury.plugin.core.TreasuryPlugin;
+import me.lokka30.treasury.plugin.core.utils.QuickTimer;
+import org.apache.logging.log4j.Logger;
+import org.spongepowered.api.Server;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.command.Command;
+import org.spongepowered.api.config.ConfigDir;
+import org.spongepowered.api.event.EventListenerRegistration;
+import org.spongepowered.api.event.Listener;
+import org.spongepowered.api.event.lifecycle.RegisterCommandEvent;
+import org.spongepowered.api.event.lifecycle.StartedEngineEvent;
+import org.spongepowered.api.event.lifecycle.StoppingEngineEvent;
+import org.spongepowered.api.event.network.ServerSideConnectionEvent;
+import org.spongepowered.plugin.PluginContainer;
+import org.spongepowered.plugin.builtin.jvm.Plugin;
+
+@Plugin("treasury")
+public class TreasurySponge {
+
+    private final Logger logger;
+    private final PluginContainer container;
+
+    @Inject
+    @ConfigDir(sharedRoot = false)
+    private Path configDirectory;
+
+    @Inject
+    TreasurySponge(PluginContainer container, Logger logger) {
+        this.logger = logger;
+        this.container = container;
+    }
+
+    private SpongeTreasuryPlugin treasuryPlugin;
+    private TreasuryCommand command;
+
+    @Listener
+    public void onStart(StartedEngineEvent<Server> event) {
```

Comment:
There are other interfaces which will be called on different stages (or lifecycles) like this. This is called when the server is ready to use, so in your current implementation you initializes everything at once which not really harmful, but I would preferable extract this into two different stages.

In my case I would do this:
- Creating instances of classes, creating/loading configs would get into `StartingEngineEvent`
- If you want listeners to be registered, you can do it in LoadedGameEvent or in this method StartedEngineEvent, the order doesn't matter

Also, if you implements reload command, you can also add a new listener method with `RefreshGameEvent` parameter, but this event is only called when you execute `/sponge plugins reload pluginName` from game.